### PR TITLE
Issues/sql options

### PR DESCRIPTION
**Pull request by pmlopes**
This is an attempt to solve a couple of issues:

* https://github.com/vert-x3/vertx-sql-common/pull/42
* have a common set of options across all clients
* https://github.com/vert-x3/vertx-jdbc-client/issues/75
* https://github.com/vert-x3/vertx-jdbc-client/issues/43

* support returning of named generated keys
* specify fetchSize amongst others result options

### CODE REVIEW

File: src/main/java/io/vertx/ext/sql/FetchDirection.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.sql;
+
+import io.vertx.codegen.annotations.VertxGen;
+
+import java.sql.ResultSet;
+
+/**
+ * Represents a The fetch direction hint
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/sql/FetchDirection.java]: > Represents the fetch direction hint

File: src/main/java/io/vertx/ext/sql/ResultSetConcurrency.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.sql;
+
+import io.vertx.codegen.annotations.VertxGen;
+
+import java.sql.ResultSet;
+
+/**
+ * Represents a The resultset concurrency hint
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/sql/ResultSetConcurrency.java]: > Represents the resultset concurrency hint

File: src/main/java/io/vertx/ext/sql/ResultSetType.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+ * You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.sql;
+
+import io.vertx.codegen.annotations.VertxGen;
+
+import java.sql.ResultSet;
+
+/**
+ * Represents a The resultset type hint
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/sql/ResultSetType.java]: > Represents the resultset type hint

File: src/main/java/io/vertx/ext/sql/SQLConnection.java [hunk 4]
```diff
@@ -34,6 +35,16 @@
 public interface SQLConnection extends AutoCloseable {
 
   /**
+   * Sets the desired options to be applied to the connection when a statement is executed.
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/sql/SQLConnection.java]: How about adding a comment to indicate if these options should be re-applied any time a connection is acquired from the pool?

File: src/main/java/io/vertx/ext/sql/SQLConnection.java [hunk 5]
```diff
@@ -34,6 +35,18 @@
...
+   *
+   * The options should be applied on a connection basis and are not applied to new connections automatically.
```
Comment (tsegismont) [inline src/main/java/io/vertx/ext/sql/SQLConnection.java]: At the risk of sounding picky, this comment is not clear to me. Will I have to apply these options any time I acquire a connection from the pool?

Comment (pmlopes) [reply src/main/java/io/vertx/ext/sql/SQLConnection.java]: Does it feel better now? :)

**Comment by tsegismont** [comment]
Yes, me relieved! Thanks :P

2017-07-25 13:17 GMT+02:00 Paulo Lopes <notifications@github.com>:

**Pull request merged**